app/src/main/java/net/bonysoft/magicmirror/FaceRecognitionActivity.java

```diff
@@ -0,0 +1,164 @@
+    private static final float CAMERA_SOURCE_REQUESTED_FPS = 30.0f;
+    private static final int CAMERA_SOURCE_WIDTH = 640;
+    private static final int CAMERA_SOURCE_HEIGHT = 360;
+
+    private CameraSource cameraSource = null;
+    private CameraSourcePreview preview;
+
+    private SystemUIHider systemUIHider;
+    private TextView faceStatus;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_face_recognition);
+
+        faceStatus = (TextView) findViewById(R.id.status);
+        preview = (CameraSourcePreview) findViewById(R.id.preview);
+
+        systemUIHider = new SystemUIHider(findViewById(android.R.id.content));
+        keepScreenOn();
+
+        boolean hasCamera = getPackageManager().hasSystemFeature(PackageManager.FEATURE_CAMERA_FRONT);
+        if (!hasCamera) {
+            Toast.makeText(this, R.string.no_camera_available_error, Toast.LENGTH_SHORT).show();
+            finish();
+            return;
+        }
+
+        if (ActivityCompat.checkSelfPermission(this, Manifest.permission.CAMERA) != PackageManager.PERMISSION_GRANTED) {
+            ActivityCompat.requestPermissions(
+                    this,
+                    new String[]{Manifest.permission.CAMERA},
+                    CAMERA_PERMISSION_REQUEST
+            );
+        } else {
+            createCameraSource();
+        }
+
+        checkForPlayServices();
+    }
+
+    private void keepScreenOn() {
+        getWindow().addFlags(WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON);
+    }
+
+    private void createCameraSource() {
+        FaceDetector detector = new FaceDetector.Builder(this)
+                .setClassificationType(FaceDetector.ALL_CLASSIFICATIONS)
+                .build();
+
+        detector.setProcessor(
+                new MultiProcessor.Builder<>(new FaceTracker.Factory(faceListener))
+                        .build()
+        );
+
+        if (!detector.isOperational()) {
+            Toast.makeText(this, R.string.face_detection_not_available_error, Toast.LENGTH_LONG).show();
+            return;
+        }
+
+        cameraSource = new CameraSource.Builder(this, detector)
+                .setRequestedPreviewSize(CAMERA_SOURCE_WIDTH, CAMERA_SOURCE_HEIGHT)
+                .setFacing(CameraSource.CAMERA_FACING_FRONT)
+                .setRequestedFps(CAMERA_SOURCE_REQUESTED_FPS)
+                .build();
+    }
+
+    private void checkForPlayServices() {
+        int code = GoogleApiAvailability.getInstance().isGooglePlayServicesAvailable(this);
+        if (code != ConnectionResult.SUCCESS) {
+            Dialog dlg = GoogleApiAvailability.getInstance().getErrorDialog(this, code, 101);
+            dlg.show();
+        }
+    }
+
+    @Override
+    protected void onResume() {
+        super.onResume();
+        systemUIHider.hideSystemUi();
+
+        startCameraSource();
```

Comment:
This looks a bit weird. If we go through `onPause` the reference to `cameraSource` is lost (set to null). What would happen if I hit home and return to the app again? Wouldn't the camera source stop working since we are only setting a reference in `onCreate`?